mono/metadata/custom-attrs.c

```diff
@@ -79,20 +79,27 @@ lookup_custom_attr (MonoImage *image, gpointer member)
 	return res;
 }
 
-static gboolean
-custom_attr_visible (MonoImage *image, MonoReflectionCustomAttr *cattr)
+static MonoMethod*
+custom_attr_visible (MonoImage *image, MonoReflectionCustomAttrHandle cattr, MonoReflectionMethodHandle ctor_handle)
+// ctor_handle is a local to this function, allocated by its caller for efficiency.
 {
 	MONO_REQ_GC_UNSAFE_MODE;
 
+	MONO_HANDLE_GET (ctor_handle, cattr, ctor);
+	MonoMethod *ctor_method = MONO_HANDLE_GETVAL (ctor_handle, method);
+
 	/* FIXME: Need to do more checks */
-	if (cattr->ctor->method && (m_class_get_image (cattr->ctor->method->klass) != image)) {
-		int visibility = mono_class_get_flags (cattr->ctor->method->klass) & TYPE_ATTRIBUTE_VISIBILITY_MASK;
+	if (ctor_method) {
+		MonoClass *klass = ctor_method->klass;
+		if (m_class_get_image (klass) != image) {
+			int visibility = mono_class_get_flags (klass) & TYPE_ATTRIBUTE_VISIBILITY_MASK;
 
-		if ((visibility != TYPE_ATTRIBUTE_PUBLIC) && (visibility != TYPE_ATTRIBUTE_NESTED_PUBLIC))
-			return FALSE;
+			if ((visibility != TYPE_ATTRIBUTE_PUBLIC) && (visibility != TYPE_ATTRIBUTE_NESTED_PUBLIC))
+				return NULL;
+		}
 	}
 
-	return TRUE;
+	return ctor_method;
```

Comment:
This is a change. Before, if ctor->method was null, we'd return true.
However later code asserts it, so this shouldn't be a change.